multinode_integration_tests/src/masq_node_cluster.rs

```diff
@@ -77,38 +77,58 @@ impl MASQNodeCluster {
     }
 
     pub fn start_mock_node_with_real_cryptde(&mut self, ports: Vec<u16>) -> MASQMockNode {
-        self.start_mock_node(ports, None)
+        let unfinished_node = self.start_configurable_mock_node(ports, None);
+        self.finish_mock_node(unfinished_node)
     }
 
     pub fn start_mock_node_with_public_key(
         &mut self,
         ports: Vec<u16>,
         public_key: &PublicKey,
     ) -> MASQMockNode {
-        self.start_mock_node(ports, Some(public_key))
+        let unfinished_node = self.start_configurable_mock_node(ports, Some(public_key));
+        self.finish_mock_node(unfinished_node)
     }
 
-    fn start_mock_node(
+    pub fn start_configurable_mock_node_with_public_key(
+        &mut self,
+        ports: Vec<u16>,
+        public_key: &PublicKey,
+    ) -> ConfigurableMASQMockNode {
+        self.start_configurable_mock_node(ports, Some(public_key))
+    }
+
+    fn start_configurable_mock_node(
         &mut self,
         ports: Vec<u16>,
         public_key_opt: Option<&PublicKey>,
-    ) -> MASQMockNode {
+    ) -> ConfigurableMASQMockNode {
         let index = self.next_index;
         self.next_index += 1;
-        let node = match public_key_opt {
-            Some(public_key) => MASQMockNode::start_with_public_key(
+        match public_key_opt {
+            Some(public_key) => ConfigurableMASQMockNode::start_with_public_key(
                 ports,
                 index,
                 self.host_node_parent_dir.clone(),
                 public_key,
                 self.chain,
             ),
-            None => {
-                MASQMockNode::start(ports, index, self.host_node_parent_dir.clone(), self.chain)
-            }
-        };
-        let name = node.name().to_string();
-        self.mock_nodes.insert(name.clone(), node);
+            None => ConfigurableMASQMockNode::start(
+                ports,
+                index,
+                self.host_node_parent_dir.clone(),
+                self.chain,
+            ),
+        }
+    }
+
+    pub fn finish_mock_node(
```

Comment:
Could you name this something that makes it clear that what we're finishing is the _configuration_ of a mock Node? We may not be finished with the mock Node itself.